### DISCUSSION TITLE
Saleor App + Webhooks per channel?

### DISCUSSION THREAD

**Discussion by JannikZed** [saleor/saleor#8207] [Q&A]
Hi there,
Saleor plugins are already configured per channel, which is nice. Is it planned, to do the same with external apps? So that an app knows already during the installation process, to which channel it is bound? And that webhooks are only sent for that certain channel as well? 
We are trying to create a nice app-base that should be able to handle all those kinds of possibilities and right now it is hard to integrate channels and apps nicely.

**Comment by karolkielecki** [comment]
@JannikZed Hi Jannik! While writing a new app, there is a setting where you can configure the app per channel. Let us know if you have more questions!

**Comment by JannikZed** [comment]
Hi Karol,
Where exactly is a setting like this? The graphql mutation to add an app has no setting for channel, the webhook create mutation as well..

**Reply by korycins** [reply]
Hey @JannikZed ,
To clarify Karol's answer. For now, you can build an App where you can have a setting for handling the channel-specific payload. 
Payloads like order or checkout have a field for channel slug. We don't have a specific configuration webhook+channel as not all webhook events relay on channel
